wallet/wallet.go

```diff
@@ -2554,6 +2560,86 @@ func (w *Wallet) GetTransaction(txHash chainhash.Hash) (*GetTransactionResult,
 	if err != nil {
 		return nil, err
 	}
+
+	// If the transaction was not found in the transaction store, we need to
+	// build the TransactionSummary from the backend RPC response.
+	if summary == nil {
+		chainClient, err := w.requireChainClient()
+		if err != nil {
+			return nil, err
+		}
+
+		// Get the transaction data from the backend endpoint.
+		var txResult *btcjson.TxRawResult
+		switch client := chainClient.(type) {
+		case *chain.RPCClient:
+			txResult, err = client.GetRawTransactionVerbose(&txHash)
+			if err != nil {
+				return nil, err
+			}
+		case *chain.BitcoindClient:
+			txResult, err = client.GetRawTransactionVerbose(&txHash)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		txRaw, err := hex.DecodeString(txResult.Hex)
+		if err != nil {
+			return nil, err
+		}
+
+		summary = &TransactionSummary{
+			Hash:        &txHash,
+			Transaction: txRaw,
+			MyInputs:    make([]TransactionSummaryInput, 0),
+			MyOutputs:   make([]TransactionSummaryOutput, 0),
+			Timestamp:   txResult.Time,
+		}
+
+		// For a confirmed transaction we must decode the block hash and
+		// get the block height.
+		if txResult.BlockHash != "" {
+			blockHash, err = chainhash.NewHashFromStr(
+				txResult.BlockHash,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			// Obtain the block height from the backend.
+			switch client := chainClient.(type) {
+			case *chain.RPCClient:
+				header, err := client.GetBlockHeaderVerbose(
+					blockHash,
+				)
+				if err != nil {
+					return nil, err
+				}
+				bestHeight = header.Height
+			case *chain.BitcoindClient:
+				bestHeight, err = client.GetBlockHeight(
+					blockHash,
+				)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+
+		// Set the TransactionSummary in the GetTransactionResult.
+		res.Summary = *summary
+		res.Height = bestHeight
+		res.BlockHash = summary.Hash
+		res.Timestamp = summary.Timestamp
+
+		// Calculate the number of confirmations.
+		if res.Height != -1 {
+			res.Confirmations = bestHeight - res.Height + 1
```

Comment:
I don't think this is correct. `bestHeight` and `res.Height` are both the exact same value at this point.

Reply:
See line 2639.

Reply:
lol. I've updated this to fetch the chain tip and then compare the heights to get the number of confirmation.